packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -908,6 +908,11 @@ class RenderOpacity extends RenderProxyBox {
     }
   }
 
+  @override
+  bool hitTestChildren(BoxHitTestResult result, {required Offset position}) {
+    return _alpha != 0 && super.hitTestChildren(result, position: position);
```

Comment:
I believe `maintainInteractivity ` of the Visibility widget (which uses an Opacity widget to implement some of its features) relies on being able to hit test fully transparent widgets: https://master-api.flutter.dev/flutter/widgets/Visibility/maintainInteractivity.html

Hopefully we have a test covering that, if not, we should add one.

Reply:
Yeah I think one of the failing tests is in visibility_tests and seems to be testing exactly that. Does it make sense to add an additional parameter on the render object and make the "hide the children when invisible" behavior opt-out?

Reply:
Is there a case we that we don't want something to be painted but it needs to remain hittestable?